### PR TITLE
Changes to consider the certify form state for enabling/disabling fil…

### PR DESCRIPTION
**Pull request by leksmall**
*Issue #:* /bcgov/entity#2129

*Description of changes:*
Changes to consider the certify form state for enabling/disabling file and pay button
Changes to the revised certify clause for BComp
Added more tests to actions
Changes to set the certify resource based on changes in entity type

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -140,5 +137,11 @@ export default class App extends Mixins(DateMixin) {
   private onRouteChanged (): void {
     this.setCurrentStep(this.$route.meta.step)
   }
+
+  @Watch('tombStoneModel.entityType')
+  private onEntityTypeChanged (val:string | null) : void{
+    this.setCertifyStatementResource(val ? CertifyStatementResource
+      .find(x => x.entityType === val) : null)
+  }
```
Comment (cameron-freshworks) [inline src/App.vue]: This is interesting! I can see this watcher ( or something similar) being expanded in the future to cover more than just the certify statement resource but also dictate what type of steps an entity has or anything else entity specific that could be handle by a resource file. Very cool.

File: src/App.vue [hunk 2]
```diff
@@ -96,7 +96,6 @@ export default class App extends Mixins(DateMixin) {
 
   private filingData: Array<FilingDataIF> = []
   private totalFee: number = 0
-  private entityType: string = 'CP'
 
   @Action('setEntityType') setEntityType!: ActionBindingIF
```
Comment (cameron-freshworks) [inline src/App.vue]: I know you didn't work on this in this PR but are we using this Binding in this component?

Comment (leksmall) [reply src/App.vue]: Good catch! Will remove it. Thanks

File: src/store/getters/state-getters.ts [hunk 3]
```diff
@@ -32,7 +32,8 @@ export const isEnableFilePayBtn = (state: any): boolean => {
   const step3Valid = true // FOR TESTING ONLY
   const step4Valid = true // FOR TESTING ONLY
   const step5Valid = true // FOR TESTING ONLY
-  return (step1Valid && step2Valid && step3Valid && step4Valid && step5Valid)
+  return (step1Valid && step2Valid && step3Valid && step4Valid && step5Valid &&
+    state.stateModel.certifyState.certifyFormValid)
```
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts]: I agree to leave this in as "for testing" for now. Ultimately "certifyFormValid" state should roll up to "step5Valid" state.

**Pull request merged**